src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -975,12 +1391,80 @@ export class SavedObjectsRepository {
     const savedObject = this._serializer.rawToSavedObject(raw);
     return omit(savedObject, 'namespace');
   }
+
+  private _rawInNamespaces(raw: SavedObjectsRawDoc, namespace?: string) {
+    const rawDocType = raw._source.type as string;
+
+    // if the type is namespace isolated, or namespace agnostic, we can continue to rely on the guarantees
+    // of the document ID format and don't need to check this
+    if (!this._registry.isMultiNamespace(rawDocType)) {
+      return true;
+    }
+
+    const namespaces = raw._source.namespaces as string[] | undefined;
+    return namespaces?.includes(getNamespaceString(namespace)) ?? false;
+  }
+
+  /**
+   * Pre-flight check to get a multi-namespace saved object's included namespaces. This ensures that, if the saved object exists, it
+   * includes the target namespace.
+   *
+   * @param type The type of the saved object.
+   * @param id The ID of the saved object.
+   * @param namespace The target namespace.
+   * @returns Array of namespaces that this saved object currently includes, or (if the object does not exist yet) the namespaces that a
+   * newly-created object will include.
+   * @throws Will throw an error if the saved object exists and it does not include the target namespace.
+   */
+  private async preflightGetNamespaces(type: string, id: string, namespace?: string) {
+    const response = await this._callCluster('get', {
```

Comment:
I didn't add a sanity check originally because I won't be able to create a unit test for it without exposing the method (which we shouldn't need to do). I'll put it in anyway, though.